test/smoke/FirefoxPrivateVPNUITest/FirefoxPrivateVPNUITest/Common/UserCommonOperation.cs

```diff
@@ -81,7 +85,129 @@ public static IRestResponse AmIMullvad()
+            Assert.AreEqual("VPN is on", mainScreen.GetTitle());
+            Assert.IsTrue(mainScreen.GetSubtitle().Contains("Secure and protected"));
+
+            // Verify user is connected to Mullvad VPN
+            response = AmIMullvad();
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.IsTrue(response.Content.Contains("You are connected to Mullvad"));
+        }
+
+        /// <summary>
+        /// User click the toggle switch to turn off VPN.
+        /// </summary>
+        /// <param name="vpnClient">VPN session.</param>
+        /// <param name="desktop">Desktop session.</param>
+        public static void DisconnectVPN(FirefoxPrivateVPNSession vpnClient, DesktopSession desktop)
+        {
+            // Verify user is already connected to VPN
+            IRestResponse response = AmIMullvad();
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.IsTrue(response.Content.Contains("You are connected to Mullvad"));
+
+            // Click VPN switch toggle and turn off VPN
+            vpnClient.Session.SwitchTo();
+            MainScreen mainScreen = new MainScreen(vpnClient.Session);
+            mainScreen.ToggleVPNSwitch();
+            Assert.IsTrue(mainScreen.GetOffImage().Displayed);
+            Assert.IsFalse(mainScreen.GetOnImage().Displayed);
+
+            // Verify the windows notification
+            Thread.Sleep(TimeSpan.FromSeconds(2));
+            desktop.Session.SwitchTo();
+            WindowsNotificationScreen windowsNotificationScreen = new WindowsNotificationScreen(desktop.Session);
+            Assert.AreEqual("VPN is off", windowsNotificationScreen.GetTitleText());
+            Assert.AreEqual("You disconnected.", windowsNotificationScreen.GetMessageText());
+            windowsNotificationScreen.ClickDismissButton();
+
+            vpnClient.Session.SwitchTo();
+            Assert.AreEqual("VPN is off", mainScreen.GetTitle());
+            Assert.AreEqual("Turn it on to protect your entire device", mainScreen.GetSubtitle());
+
+            // Verify user disconnected to Mullvad VPN
+            response = AmIMullvad();
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.IsTrue(response.Content.Contains("You are not connected to Mullvad"));
+        }
+
+        /// <summary>
+        /// Execute the request with retry policy.
+        /// </summary>
+        /// <param name="client">RestClient object.</param>
+        /// <param name="request">RestRequest object.</param>
+        /// <returns>The API response.</returns>
+        private static IRestResponse RetryExecute(IRestClient client, IRestRequest request)
+        {
+            // Define retry policy
+            var policy = Policy.HandleResult<IRestResponse>((response) =>
+            {
+                return response.StatusCode != HttpStatusCode.OK;
+            }).WaitAndRetry(3, (count) => TimeSpan.FromSeconds(10));
```

Comment:
Is the number of retry attempts (3) and wait period (10 s) here arbitrary?  How was this determined?

Reply:
It is arbitrary. Should we move these to env variable? I am thinking maybe we have too many env var.